HPXMLtoOpenStudio/resources/vehicle.rb

```diff
@@ -0,0 +1,238 @@
+# frozen_string_literal: true
+
+# Collection of methods for adding vehicle-related OpenStudio objects, built on the Battery class
+class Vehicle
+  # Adds any HPXML Vehicles to the OpenStudio model.
+  # Currently only models electric vehicles.
+  #
+  # @param runner [OpenStudio::Measure::OSRunner] Object typically used to display warnings
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param spaces [Hash] Map of HPXML locations => OpenStudio Space objects
+  # @param hpxml_bldg [HPXML::Building] HPXML Building object representing an individual dwelling unit
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [nil]
+  def self.apply(runner, model, spaces, hpxml_bldg, schedules_file)
+    hpxml_bldg.vehicles.each do |vehicle|
+      next unless vehicle.vehicle_type == Constants::ObjectTypeBatteryElectricVehicle
+
+      apply_electric_vehicle(runner, model, spaces, hpxml_bldg, vehicle, schedules_file)
+    end
+  end
+
+  # Retrieves EV charging and discharging schedules
+  #
+  # @param runner [OpenStudio::Measure::OSRunner] Object typically used to display warnings
+  # @param model [OpenStudio::Model::Model] OpenStudio Model object
+  # @param vehicle [HPXML::Vehicle] Object that defines a single electric vehicle
+  # @param schedules_file [SchedulesFile] SchedulesFile wrapper class instance of detailed schedule files
+  # @return [Array<OpenStudio::Model::ScheduleRuleset or OpenStudio::Model::ScheduleFile>] The charging and discharging schedules, either as a ScheduleRuleset or as a ScheduleFile
+  def self.get_ev_charging_schedules(runner, model, vehicle, schedules_file)
```

Comment:
This method was created so that the discharge schedule could be accessed and the rated power output could be set prior to the Battery.apply call. The process looks like:
- call get_ev_charging_schedules() to get discharge schedule
- parse discharge schedule and sum the annual hours
- calculate effective discharge power based on annual driving hours, efficiency, and annual miles
- estimate the rated power output using the effective discharge * 2.25
- apply battery (which does call get_ev_charging_schedules() once more)